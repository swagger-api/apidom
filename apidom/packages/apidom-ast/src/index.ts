// JSON AST related exports
export { default as JsonNode } from './json/nodes/JsonNode';
export { default as JsonDocument } from './json/nodes/JsonDocument';
export { default as JsonObject } from './json/nodes/JsonObject';
export { default as JsonProperty } from './json/nodes/JsonProperty';
export { default as JsonArray } from './json/nodes/JsonArray';
export { default as JsonValue } from './json/nodes/JsonValue';
export { default as JsonKey } from './json/nodes/JsonKey';
export { default as JsonString } from './json/nodes/JsonString';
export { default as JsonStringContent } from './json/nodes/JsonStringContent';
export { default as JsonEscapeSequence } from './json/nodes/JsonEscapeSequence';
export { default as JsonNumber } from './json/nodes/JsonNumber';
export { default as JsonTrue } from './json/nodes/JsonTrue';
export { default as JsonFalse } from './json/nodes/JsonFalse';
export { default as JsonNull } from './json/nodes/JsonNull';
export {
  isFalse as isJsonFalse,
  isProperty as isJsonProperty,
  isStringContent as isJsonStringContent,
  isEscapeSequence as isJsonEscapeSequence,
  isArray as isJsonArray,
  isKey as isJsonKey,
  isNull as isJsonNull,
  isNumber as isJsonNumber,
  isObject as isJsonObject,
  isString as isJsonString,
  isTrue as isJsonTrue,
} from './json/nodes/predicates';
// YAML AST related exports
export { default as YamlAlias } from './yaml/nodes/YamlAlias';
export { default as YamlCollection } from './yaml/nodes/YamlCollection';
export { default as YamlComment } from './yaml/nodes/YamlComment';
export { default as YamlDirective } from './yaml/nodes/YamlDirective';
export { default as YamlDocument } from './yaml/nodes/YamlDocument';
export { default as YamlKeyValuePair } from './yaml/nodes/YamlKeyValuePair';
export { default as YamlMapping } from './yaml/nodes/YamlMapping';
export { default as YamlNode } from './yaml/nodes/YamlNode';
export { default as YamlScalar } from './yaml/nodes/YamlScalar';
export { default as YamlSequence } from './yaml/nodes/YamlSequence';
export { default as YamlStream } from './yaml/nodes/YamlStream';
export { default as YamlTag } from './yaml/nodes/YamlTag';
export { default as YamlAnchor } from './yaml/nodes/YamlAnchor';
export {
  isAlias as isYamlAlias,
  isKeyValuePair as isYamlKeyValuePair,
  isDirective as isYamlDirective,
  isDocument as isYamlDocument,
  isMapping as isYamlMapping,
  isScalar as isYamlScalar,
  isSequence as isYamlSequence,
  isStream as isYamlStream,
  isTag as isYamlTag,
} from './yaml/nodes/predicates';
// generic AST related exports
export { default as Literal } from './Literal';
export { Point, default as Position } from './Position';
export { default as Error } from './Error';
export { default as ParseResult } from './ParseResult';
// AST traversal related exports
export { getVisitFn, BREAK, visit } from './visitor';
// YAML CST/AST transformers related exports
export {
  transform as transformTreeSitterYamlCST,
  keyMap as treeSitterYamlKeyMap,
} from './yaml/transformers/tree-sitter-yaml';
