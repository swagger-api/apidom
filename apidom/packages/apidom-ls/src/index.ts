// @ts-ignore
import ApiDOMParser from 'apidom-parser';
// @ts-ignore
import * as openapi3_1Adapter from 'apidom-parser-adapter-openapi-json-3-1';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default as getLanguageService } from './apidom-language-service';

export type {
  LanguageService,
  LanguageSettings,
  SymbolsContext,
  ValidationContext,
  CompletionContext,
  WorkspaceContextService,
  ColorsContext,
  LanguageServiceContext,
  SUPPORTED_LANGUAGES,
} from './apidom-language-types';

export const pocTODOToRemove: () => Promise<string> = async () => {
  const parser = ApiDOMParser();

  const value = `{
    "openapi": "3.0.0",
      "info": {
        "version": "0.1.9"
      }
    }`;
  parser.use(openapi3_1Adapter);
  // parser.use(asyncapi2_0Adapter);
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const parseResult = await parser.parse(value, { sourceMap: true });
  // console.log('Par', JSON.stringify(parseResult));
  return 'done';
};
