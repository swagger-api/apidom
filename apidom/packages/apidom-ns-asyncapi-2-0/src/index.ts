import './refractor/registration';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default } from './namespace';

export {
  isAsyncApi2_0Element,
  isAsyncApiVersionElement,
  isChannelBindingsElement,
  isChannelItemElement,
  isChannelsElement,
  isComponentsElement,
  isContactElement,
  isIdentifierElement,
  isInfoElement,
  isLicenseElement,
  isOperationElement,
  isParameterElement,
  isParametersElement,
  isReferenceElement,
  isReferenceElementExternal,
  isSchemaElement,
  isSecurityRequirementElement,
  isServerElement,
  isServerBindingsElement,
  isServersElement,
  isServerVariableElement,
} from './predicates';

export {
  isAsyncApi2_0LikeElement,
  isParameterLikeElement,
  isReferenceLikeElement,
  isSchemaLikeElement,
  isSecurityRequirementLikeElement,
  isServerLikeElement,
  isAsyncApiExtension,
} from './refractor/predicates';

export { keyMap, getNodeType } from './traversal/visitor';

export { default as AsyncApi2_0Element } from './elements/AsyncApi2-0';
export { default as AsyncApiVersionElement } from './elements/AsyncApiVersion';
export { default as ChannelBindingsElement } from './elements/ChannelBindings';
export { default as ChannelItemElement } from './elements/ChannelItem';
export { default as ChannelsElement } from './elements/Channels';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as CorrelationIDElement } from './elements/CorrelationID';
export { default as DefaultContentTypeElement } from './elements/DefaultContentType';
export { default as IdentifierElement } from './elements/Identifier';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as OperationElement } from './elements/Operation';
export { default as ParameterElement } from './elements/Parameter';
export { default as ParametersElement } from './elements/Parameters';
export { default as ReferenceElement } from './elements/Reference';
export { default as SchemaElement } from './elements/Schema';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerElement } from './elements/Server';
export { default as ServerBindingsElement } from './elements/ServerBindings';
export { default as ServersElement } from './elements/Servers';
export { default as ServerVariableElement } from './elements/ServerVariable';
