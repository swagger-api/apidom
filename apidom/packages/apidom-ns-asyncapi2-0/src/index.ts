export { default } from './namespace';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export {
  isAsycApi2_0Element,
  isAsycapiElement,
  isComponentsElement,
  isContactElement,
  isIdentifierElement,
  isInfoElement,
  isLicenseElement,
  isSchemaElement,
  isChannelsElement,
  isChannelItemElement,
  isServersElement,
  isServerElement,
} from './predicates';

export { default as AsyncapiElement } from './elements/Asyncapi';
export { default as AsyncApi2_0Element } from './elements/AsyncApi2-0';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as IdentifierElement } from './elements/Identifier';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as SchemaElement } from './elements/Schema';
export { default as ChannelsElement } from './elements/Channels';
export { default as ChannelItemElement } from './elements/ChannelItem';
export { default as OperationElement } from './elements/Operation';
export { default as ParametersElement } from './elements/Parameters';
export { default as ChannelBindingsElement } from './elements/ChannelBindings';
export { default as ServersElement } from './elements/Servers';
export { default as ServerElement } from './elements/Server';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerBindingsElement } from './elements/ServerBindings';
