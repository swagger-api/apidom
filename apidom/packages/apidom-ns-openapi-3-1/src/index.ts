import './refractor/registration';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default } from './namespace';

export {
  isCallbackElement,
  isComponentsElement,
  isContactElement,
  isExternalDocumentationElement,
  isInfoElement,
  isJsonSchemaDialectElement,
  isLicenseElement,
  isOpenapiElement,
  isOpenApi3_1Element,
  isOperationElement,
  isParameterElement,
  isPathItemElement,
  isPathsElement,
  isReferenceElement,
  isReferenceElementExternal,
  isRequestBodyElement,
  isResponseElement,
  isResponsesElement,
  isSchemaElement,
  isSchemaElementExternal,
  isSecurityRequirementElement,
  isServerElement,
  isServerVariableElement,
  isMediaTypeElement,
} from './predicates';

export {
  isOpenApi3_1LikeElement,
  isParameterLikeElement,
  isReferenceLikeElement,
  isRequestBodyLikeElement,
  isResponseLikeElement,
  isServerLikeElement,
  isOpenApiExtension,
} from './refractor/predicates';

export { keyMap, getNodeType } from './traversal/visitor';

export { default as CallbackElement } from './elements/Callback';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as DiscriminatorElement } from './elements/Discriminator';
export { default as ExternalDocumentationElement } from './elements/ExternalDocumentation';
export { default as InfoElement } from './elements/Info';
export { default as JsonSchemaDialectVisitor } from './elements/JsonSchemaDialect';
export { default as LicenseElement } from './elements/License';
export { default as MediaTypeElement } from './elements/MediaType';
export { default as OpenapiElement } from './elements/Openapi';
export { default as OpenApi3_1Element } from './elements/OpenApi3-1';
export { default as OperationElement } from './elements/Operation';
export { default as ParameterElement } from './elements/Parameter';
export { default as PathItemElement } from './elements/PathItem';
export { default as PathsElement } from './elements/Paths';
export { default as ReferenceElement } from './elements/Reference';
export { default as RequestBodyElement } from './elements/RequestBody';
export { default as ResponseElement } from './elements/Response';
export { default as ResponsesElement } from './elements/Responses';
export { default as SchemaElement } from './elements/Schema';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerElement } from './elements/Server';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as XmlElement } from './elements/Xml';
