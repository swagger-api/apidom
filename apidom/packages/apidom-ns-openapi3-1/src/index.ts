export { default } from './namespace';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export {
  isOpenApiApi3_1Element,
  isContactElement,
  isLicenseElement,
  isInfoElement,
  isComponentsElement,
  isSchemaElement,
  isOpenapiElement,
  isServerElement,
  isServerVariableElement,
  isPathsElement,
  isPathItemElement,
  isOperationElement,
} from './predicates';

export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as OpenapiElement } from './elements/Openapi';
export { default as OpenApi3_1Element } from './elements/OpenApi3-1';
export { default as SchemaElement } from './elements/Schema';
export { default as ServerElement } from './elements/Server';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as PathsElement } from './elements/Paths';
export { default as PathItemElement } from './elements/PathItem';
export { default as OperationElement } from './elements/Operation';
export { default as ParameterElement } from './elements/Parameter';
