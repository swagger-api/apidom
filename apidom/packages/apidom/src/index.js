'use strict';

const defaultNamespace = require('./namespace');

const createNamespace = (namespacePlugin) => {
  const namespace = new defaultNamespace.Namespace()
  namespace.use(namespacePlugin);
  return namespace;
};

// toJSON :: (Namespace, Element) -> Object
const toJSON = (namespace, element) => namespace.toRefract(element);

// toJSONString :: (Namespace, Element) -> String
const toJSONString = (namespace, element) => JSON.stringify(toJSON(namespace, element));

// fromJSON :: (Namespace, Object) -> Element
const fromJSON = (namespace, json) => namespace.fromRefract(json);

// fromJSONString :: (Namespace, String) -> Element
const fromJSONString = (namespace, jsonString) => fromJSON(JSON.parse(jsonString));

// Reconstructs the ApiDOM into JavaScript POJO.
// This POJO would be  the result of parsing the original
// JSON string with JSON.parse function.
// toValue :: Element -> Object
const toValue = element => element.toValue();

module.exports = {
  namespace: defaultNamespace,
  createNamespace,
  toJSON,
  toJSONString,
  fromJSON,
  fromJSONString,
  toValue,
}
