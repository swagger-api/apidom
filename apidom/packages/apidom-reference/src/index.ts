import { ParseResultElement, Element } from 'apidom';

import File from './util/File';
import * as url from './util/url';
import { ReferenceSet as IReferenceSet } from './types';
import defaultOptions from './options';
import { merge as mergeOptions } from './options/util';
import parseFn, { readFile as readFileFn } from './parse';
import resolveFn, { resolveApiDOM as resolveApiDOMFn } from './resolve';
import { dereferenceApiDOM as dereferenceApiDOMFn } from './dereference';

export const readFile = async (uri: string, options = {}): Promise<Buffer> => {
  const mergedOptions = mergeOptions(defaultOptions, options);
  const sanitizedURI = url.isFileSystemPath(uri) ? url.fromFileSystemPath(uri) : uri;
  const file = File({ uri: sanitizedURI });

  return readFileFn(file, mergedOptions);
};

export const parse = async (uri: string, options = {}): Promise<ParseResultElement> => {
  const mergedOptions = mergeOptions(defaultOptions, options);
  return parseFn(uri, mergedOptions);
};

export const resolve = async (uri: string, options = {}): Promise<IReferenceSet> => {
  const mergedOptions = mergeOptions(defaultOptions, options);
  return resolveFn(uri, mergedOptions);
};

export const resolveApiDOM = async <T extends Element>(
  element: T,
  options = {},
): Promise<IReferenceSet> => {
  const mergedOptions = mergeOptions(defaultOptions, options);
  return resolveApiDOMFn(element, mergedOptions);
};

export const dereferenceApiDOM = async <T extends Element>(
  element: T,
  options = {},
): Promise<T> => {
  const mergedOptions = mergeOptions(defaultOptions, options);
  return dereferenceApiDOMFn(element, mergedOptions);
};
