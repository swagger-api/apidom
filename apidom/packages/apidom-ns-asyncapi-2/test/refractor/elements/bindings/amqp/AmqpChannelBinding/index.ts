import { expect } from 'chai';
import { sexprs } from 'apidom';

import { AmqpChannelBindingElement } from '../../../../../../src';

describe('refractor', function () {
  context('elements', function () {
    context('AmqpChannelBindingElement', function () {
      specify('should refract to semantic ApiDOM tree', function () {
        const amqpChannelBindingElement = AmqpChannelBindingElement.refract({
          is: 'routingKey',
          exchange: {
            name: 'myExchange',
            type: 'topic',
            durable: true,
            autoDelete: false,
            vhost: '/',
          },
          queue: {
            name: 'my-queue-name',
            durable: true,
            exclusive: true,
            autoDelete: false,
            vhost: '/',
          },
          bindingVersion: '0.2.0',
        });

        expect(sexprs(amqpChannelBindingElement)).toMatchSnapshot();
      });
    });
  });
});
