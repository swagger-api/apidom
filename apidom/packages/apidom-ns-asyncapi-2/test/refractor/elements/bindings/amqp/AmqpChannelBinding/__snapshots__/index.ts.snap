// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refractor elements AmqpChannelBindingElement should refract to semantic ApiDOM tree 1`] = `
(AmqpChannelBindingElement
  (MemberElement
    (StringElement)
    (StringElement))
  (MemberElement
    (StringElement)
    (ObjectElement
      (MemberElement
        (StringElement)
        (StringElement))
      (MemberElement
        (StringElement)
        (StringElement))
      (MemberElement
        (StringElement)
        (BooleanElement))
      (MemberElement
        (StringElement)
        (BooleanElement))
      (MemberElement
        (StringElement)
        (StringElement))))
  (MemberElement
    (StringElement)
    (ObjectElement
      (MemberElement
        (StringElement)
        (StringElement))
      (MemberElement
        (StringElement)
        (BooleanElement))
      (MemberElement
        (StringElement)
        (BooleanElement))
      (MemberElement
        (StringElement)
        (BooleanElement))
      (MemberElement
        (StringElement)
        (StringElement))))
  (MemberElement
    (StringElement)
    (StringElement)))
`;
